### ISSUE TITLE
Seems like quantificator symbol do not works correctly in macros expression

### ISSUE THREAD

**Issue by sergio-ivanuzzo** [rust-lang/rust#103256]
I tried this code:

```rust
#[macro_export]
macro_rules! vec2 {
    ( $( $x:expr ),* ) => {
        {
            let mut temp_vec = Vec::new();
            $(
                temp_vec.push($x);
            )? // this one works the same as below
            temp_vec
        }
    };
}

#[macro_export]
macro_rules! vec3 {
    ( $( $x:expr ),* ) => {
        {
            let mut temp_vec = Vec::new();
            $(
                temp_vec.push($x);
            )* // this one works the same as above
            temp_vec
        }
    };
}
```

I expected to see this happen: *macros expression that use "?" should return error*

Instead, this happened: *both macros works in same way*

### Meta

`rustc --version --verbose`:
```
rustc 1.60.0 (7737e0b5c 2022-04-04)
binary: rustc
commit-hash: 7737e0b5c4103216d6fd8cf941b7ab9bdbaace7c
commit-date: 2022-04-04
host: x86_64-pc-windows-msvc
release: 1.60.0
LLVM version: 14.0.0

```

**Comment by PatchMixolydic** [comment]
This is caught by the allow-by-default [`meta_variable_misuse`](https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#meta-variable-misuse) lint ([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=02d7058d542d8821a666c598b75b6edb)):
```rust
warning: meta-variable repeats with different Kleene operator
  --> src/lib.rs:9:31
   |
5  |     ( $( $x:expr ),* ) => {
   |                    - expected repetition
...
9  |                 temp_vec.push($x);
   |                               ^^
10 |             )? // this one works the same as below
   |              - conflicting repetition
   |
note: the lint level is defined here
  --> src/lib.rs:1:9
   |
1  | #![warn(meta_variable_misuse)]
   |         ^^^^^^^^^^^^^^^^^^^^
```

See also #61053, the original tracking issue for the lint, and #95943, which contains a discussion on why this lint is allow-by-default.

**Comment by PatchMixolydic** [comment]
As far as I'm aware, you can leave it turned on as long as you're willing to accept a few false positives. For what its worth, I don't remember encountering any false positives, but this lint has saved me a bit of trouble several times.

**Comment by fmease** [comment]
Some discussion happened here: [Zulip | #wg-macros > Repetition operator mismatch between transcriber and matcher](https://rust-lang.zulipchat.com/#narrow/stream/404510-wg-macros/topic/Repetition.20operator.20mismatch.20between.20transcriber.20and.20matcher).
